pkg/service/auth.go

```diff
@@ -47,8 +48,15 @@ func (self *AuthService) Check(ctx context.Context, req *envoy_auth.CheckRequest
 	}
 	authServiceLog.Info("Check()", "reqJSON", string(reqJSON))
 
-	// service config
-	host := req.Attributes.Request.Http.Host
+	// If the host header contains a port, remove it, we only want to match the host part.
+	var host string
+	if strings.Contains(req.Attributes.Request.Http.Host,":") {
```

Comment:
Should we avoid caching the port number as well (if included in `hosts`)?

Reply:
yup, I will add it to this PR.